clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -990,16 +990,17 @@ UnsentRequest buildRequest() {
             }
 
             ShareAcknowledgeRequest.Builder requestBuilder = sessionHandler.newShareAcknowledgeBuilder(groupId, fetchConfig);
-            Node nodeToSend = metadata.fetch().nodeById(nodeId);
 
             log.trace("Building acknowledgements to send : {}", finalAcknowledgementsToSend);
```

Comment:
This log line probably ought to move down into the else block too.

Reply:
Yeah makes sense, thanks. I have moved it now.